project/app/views.py

```diff
@@ -1,71 +1,71 @@
-    if serializer.is_valid():  # find how we can ignore if and else here.
-        serializer.save()
-        return Response(status=status.HTTP_200_OK)
-    else:
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+class ProfileSignUpView(CreateAPIView):
+    queryset = UserProfile.objects.all()
+    serializer_class = UserProfileSerializer
 
 
-@api_view(["GET"])
-@permission_classes([IsAuthenticated])
-def getProfile(request):
-    decodedPayload = jwt.decode(request.headers["authorization"].split()[1], algorithms=['HS512'],
-                                key='django-insecure-ccy8d&94ym$t^wa4_6dx2!+u#!v=)@1g7)4lob9z8=o!trzg98')
-    user = CustomUser.objects.get(id=decodedPayload["user_id"])
-    customserializer = UserSerializer(user, many=False)
-    userrole = customserializer.data["role"]
-    if userrole == "A":
-        user = UserProfile.objects.all()
-        serializer = UserProfileSerializer(user, many=True)
-        return Response(serializer.data)
-    else:
-        return Response(status=status.HTTP_401_UNAUTHORIZED)
+class ProfileView(APIView):
+    permission_classes = [IsAuthenticated]
 
+    def get(self, request):
+        user = request.user
+        serializer = UserSerializer(user)
+        user_role = serializer.data["role"]
+        user_id = serializer.data["id"]
+        if user_role == "A":
+            user = UserProfile.objects.all()
+            serializer = UserProfileSerializer(user, many=True)
+            return Response(serializer.data)
+        else:
+            user = UserProfile.objects.get(user=user_id)
+            serializer = UserProfileSerializer(user)
+            return Response(serializer.data)
 
-@api_view(["POST"])
-def login(request):
-    serializer = LoginSerializer(data=request.data)
-    password = serializer.initial_data["password"]
-    phone = serializer.initial_data["phone_number"]
-    user = authenticate(request, phone_number=phone, password=password)
-    if user:
-        Refresh = RefreshToken.for_user(user)
-        return Response({
-            'refresh': str(Refresh),
-            "access": str(Refresh.access_token)
-        }
-        )
-    else:
-        return Response(status=status.HTTP_401_UNAUTHORIZED)
 
+class UpdateProfileView(APIView):
+    permission_classes = [IsAuthenticated]
 
-@api_view(['PUT'])
-@permission_classes([IsAuthenticated])
-def updateProfile(request):
-    user = request.user
-    serializer = UserProfileSerializer(user, data=request.data, partial=True)
-    if serializer.is_valid():
-        serializer.save()
-    return Response(serializer.data)
+    def put(self, request):
+        user = request.user
+        serializer = UserSerializer(user)
+        user_id = serializer.data["id"]
+        user_profile = UserProfile.objects.get(user_id=user_id)
+        profile_serializer = UserProfileSerializer(instance=user_profile, data=request.data)
+
+        if profile_serializer.is_valid():
```

Comment:
This is correct but can be updated like,

if not profile_serailizer.is_valid():
      return Response(status=status.HTTP_404_NOT_FOUND)
profile_serializer.save()
return Response(profile_serializer.data)

This will remove the else statement as well as one endpoint, And same for LoginView